packages/ui-components/src/utils/select.helper.ts

```diff
@@ -0,0 +1,56 @@
+import { isEmpty } from 'lodash';
+import { ISelectMultiOptions, ISelectOption } from '../types';
+
+export const getSelectableOptions = (options: ISelectMultiOptions = {}): ISelectMultiOptions => {
+	return Object.values(options).reduce<ISelectMultiOptions>((accumulator, option) => {
+		if (isEmpty(option.options)) {
+			accumulator[option.value] = option;
+		} else {
+			accumulator = {
+				...accumulator,
+				...getSelectableOptions(option.options)
+			};
+		}
+
+		return accumulator;
+	}, {});
+};
+
+export const getSelectedSelectableOptions = (options: ISelectMultiOptions = {}, selectedOptions: Record<string, boolean> = {}): ISelectMultiOptions => {
+	const selectableOptions = getSelectableOptions(options);
+
+	return Object.keys(selectedOptions).reduce((accumulator, optionKey) => {
+		const selectedOption = selectableOptions[optionKey];
+		const isOptionSelected = selectedOptions[optionKey] === true;
+
+		if (selectedOption && isOptionSelected) {
+			accumulator[optionKey] = selectedOption;
+		}
+
+		return accumulator;
+	}, {});
+};
+
+export const getFlattenSelectOptions = (selectOptions: Record<string, ISelectOption> = {}): Record<string, ISelectOption> =>
+	Object.values(selectOptions).reduce<Record<string, ISelectOption>>(
+		(accumulator, selectOption) => ({ ...accumulator, [selectOption.value]: selectOption, ...getFlattenSelectOptions(selectOption.options) }),
```

Comment:
if we have too many options this process of creating new references in each iteration has a high price.

Reply:
Will change this too.